### ISSUE TITLE
Best practices for distributed learning

### ISSUE THREAD

**Issue by Tutufa** [uber/horovod#127]
Hi, I'm trying to do distributed learning on 4 GPU's. Horovod works great, but the quality of the model I get using 4 GPU is worse than quality single GPU model. What are best practices, tips, and tricks for moving single GPU training into multi GPU?

It seems like my network oscillates and doesn't converge. How to deal with that?

**Comment by alsrgv** [comment]
@Tutufa, you can get inspiration from [this example](https://github.com/uber/horovod/blob/master/examples/keras_mnist_advanced.py):

1. Adjust learning rate: `lr = single_gpu_lr * num_gpus`.
2. Use learning rate warm-up: see `hvd.callbacks.LearningRateWarmupCallback`.
3. Don't forget to use `hvd.callbacks.BroadcastGlobalVariablesCallback` to make sure all workers initialize from a consistent state.
